### PR TITLE
cran release, upversion to 0.4.2 [skip vbump]

### LINKED ISSUE #212: [Release]: teal.widgets 0.4.2
### Blocked by

Nothing is blocking this release.

### Pre-requisites

- [x] Make sure that high priority bugs (label "priority" + "bug") have been resolved before going into the release.
- [ ] Review old/hanging PRs before going into the release (Optional).
- [ ] Revisit R-package's lifecycle badges (Optional).
- [x] Make sure that all upstream dependencies of this package that need to be submitted to CRAN were accepted before going into release activities.
- [x] Make sure integration tests are green 2-3 days before the release. Look carefully through logs (check for warnings and notes).
- [x] Decide what gets merged in before starting release activities.

### Release Checklist

- [x] Update NEWS.md file: make sure it reflects a holistic summary of what has changed in the package.
- [x] Remove the additional fields (`Remotes` and `Config/Needs/*`) from the DESCRIPTION file where applicable. (DU: irrelevant)
- [x] Increase versioned dependency on {package name} to >=X.X.X (Optional).
- [x] Make sure that the minimum dependency versions are updated in the DESCRIPTION file for the package and its reverse dependencies (Optional).
- [x] Create a pull request to make necessary bug fixes/changes (add "[skip vbump]" in the pr title), and after merging the PR, tag the update(s) as a release candidate v < intended release version > -rc < release candidate iteration > on the main branch.
- [x] Build the package locally using the command:`R CMD build .` which will generate a .tar.gz file necessary for the CRAN submission.
- [x] Submit the package that was build in the previous step via this form: https://cran.r-project.org/submit.html.
- [x] Address CRAN feedback, tag the package vX.X.X-rc(n+1) and repeat the submission to CRAN whenever necessary.
- [x] Get the package accepted and published on CRAN.
- [ ] If the additional fields were removed, add them back in a separate PR, and then merge the PR back to main. Note: Take precautionary measures to ensure that the version bump does not take place on a merge.
- [ ] Create a git tag with the final version set to X.X.X on the main branch.

### Testing

- [ ] Integration tests results - accepted.
- [ ] UAT results - accepted.
- [ ] All testing activities are finalized.

### Release Feedback

- [ ] Fix 1
- [ ] Enhancement 1
- [ ] Defect 1

### Post-release Checklist

- [ ] Make sure that the package is published to internal repositories.
- [ ] Review and update installation instructions for the package wherever needed (Optional).
- [ ] Update all integration tests to reference the new release.
- [ ] Ensure a new dev version (.9XXX) is added to the NEWS.md file and DESCRIPTION file as a placeholder for release notes.
- [ ] Announce the release on ________.

### Decision tree

Click [here](https://github.com/insightsengineering/.github/blob/main/.github/ISSUE_TEMPLATE/RELEASE_DECISION_TREE.md) to see the release decision tree.

### PR DESCRIPTION
**Pull request by donyunardi**
Fixes #212

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
-# teal.widgets 0.4.1.9010
+# teal.widgets 0.4.2
+
+### Enhancements
+* Introduce `nested_closeable_modal()` that can create nested popups inside an already existing popup.
```
Comment (chlebowa) [inline NEWS.md:4]: For the record, I have some doubts about this feature that I expressed after it was added and that were not addressed.

File: NEWS.md [hunk 2]
```diff
@@ -1,4 +1,9 @@
...
+* Replaces examples from `runApp` with `shinyApp`.
+* Ensure that `table_with_settings()` is synchronized with the latest update of `rtables`, and updated the version dependency accordingly.
```
[the remark this answers is not in the document]
Comment (donyunardi) [reply NEWS.md:6]: Sure.

File: NEWS.md (hunk 1 above)
Comment (donyunardi) [reply NEWS.md:4]: Can you open an issue to express your concern, ensuring that we don't overlook it? 
We added the `experimental` label to this function. If we miss anything, we can update it in an iterative manner.

Comment (chlebowa) [reply NEWS.md:4]: See [here](https://github.com/insightsengineering/teal.widgets/pull/199#issuecomment-1768133369) and below.

Comment (donyunardi) [reply NEWS.md:4]: Thanks, I created an issue for this.
https://github.com/insightsengineering/teal.widgets/issues/214

**Review by pawelru — APPROVED** [review]

**Comment by donyunardi** [comment]
Package submitted to CRAN.
Waiting for feedback.

**Pull request merged**